### PR TITLE
some doc and deprecation fixes

### PR DESCRIPTION
**Pull request by JeffBezanson**
Some stuff I came across when looking for undocumented stuff.

@stevengj I believe `besselhx` is also undocumented.

### CODE REVIEW

File: base/deprecated.jl [hunk 1]
```diff
@@ -292,36 +292,7 @@ export call
...
-    if err_stop != nothing
-        depwarn("err_stop is deprecated, use pmap(f, c...; on_error = error_handling_func).", :pmap)
-        if err_stop == false
-            kwargs[:on_error] = e->e
-        end
-    end
-
-    pmap(p, f, c...; kwargs...)
-end
+# deprecation warnings are in pmap.jl
```
Comment (tkelman) [inline base/deprecated.jl]: this is going to be really easy to overlook

File: doc/DocCheck.jl [hunk 2]
```diff
@@ -34,10 +34,9 @@ undefined_exports() = undefined(Base)
 # Check for exported names that aren't documented,
 # and return a Dict with (fn::Symbol, fullname::AbstractString) pairs
 function undocumented(m::Module)
-    init_help()
     undoc = Dict{Symbol, Array}()
     for v in sort(names(m))
-        if isdefined(m,v) && !isdocumented(v) && !isdeprecated(m,v)
+        if isdefined(m,v) && !isdocumented(eval(m,v)) && !isdeprecated(m,v)
```
Comment (tkelman) [inline doc/DocCheck.jl]: getfield ?

File: base/deprecated.jl (hunk 1 above)
Comment (JeffBezanson) [inline base/deprecated.jl]: Well, I tried. Unfortunately we simply don't have a way to deprecate a subset of keyword arguments using a separate definition.

Comment (tkelman) [inline base/deprecated.jl]: so has the deprecation been broken in its current form?

Comment (JeffBezanson) [inline base/deprecated.jl]: Yes, the deprecation did not work.

**Comment by stevengj** [comment]
`besselhx(nu,k,z)` corresponds to the Amos [zbesh](http://www.netlib.no/netlib/amos/zbesh.f) function called with `kode=2`, i.e. it computes the scaled Hankel function `exp((2k-3)*z*im) * besselh(nu, k, z)` where `k` is `1` or `2`.

**Pull request merged**